### PR TITLE
3654 - History: Add state history compare item + button ui

### PR DESCRIPTION
**Pull request by minotogna**

### CODE REVIEW

File: src/client/store/data/reducers/resetHistory.ts [hunk 1]
```diff
@@ -1,10 +1,9 @@
 import { Draft } from '@reduxjs/toolkit'
-import { Objects } from 'utils/objects'
 
-import { DataState } from 'client/store/data/stateType'
+import { DataState } from 'client/store/data/state'
 
 export const resetHistory = (state: Draft<DataState>) => {
-  Objects.unset(state.history, ['items'])
+  state.history = {}
```
Comment (sorja) [inline src/client/store/data/reducers/resetHistory.ts:6]: is this preferred over unset?

File: src/client/store/data/reducers/toggleCompareHistoryItem.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+type Action = PayloadAction<{
+  datum: ActivityLog<never>
+  target: HistoryTarget
+}>
+
+export const toggleCompareHistoryItem = (state: Draft<DataState>, action: Action) => {
+  const { datum, target } = action.payload
+
+  if (state.history.compareItem?.[target]?.id === datum.id) {
```
Comment (sorja) [inline src/client/store/data/reducers/toggleCompareHistoryItem.ts:17]: better uuid ?

File: src/client/store/data/reducers/resetHistory.ts (hunk 1 above)
Comment (minotogna) [reply src/client/store/data/reducers/resetHistory.ts:6]: It depends . Here I just need to create an empty object and this way is certainly more performant than unsetting 2 objects

File: src/client/store/data/reducers/toggleCompareHistoryItem.ts (hunk 2 above)
Comment (minotogna) [reply src/client/store/data/reducers/toggleCompareHistoryItem.ts:17]: Ok . Next pr ?

Comment (minotogna) [reply src/client/store/data/reducers/toggleCompareHistoryItem.ts:17]: Do we have uuid in activity log ?

Comment (sorja) [reply src/client/store/data/reducers/toggleCompareHistoryItem.ts:17]: If we have, check please
If not, should we migrate ?

either way, next pr is fine

**Review by sorja — APPROVED** [review]

File: src/client/store/data/reducers/toggleCompareHistoryItem.ts (hunk 2 above)
Comment (minotogna) [reply src/client/store/data/reducers/toggleCompareHistoryItem.ts:17]: I don't know if we need to migrate that . These activities are inserted very frequently and having a uuid for those might make the insert slower. Let's talk tomorrow

**Pull request merged**